operator/v1/types_etcd.go

```diff
@@ -0,0 +1,42 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Etcd provides information to configure an operator to manage kube-apiserver.
+type Etcd struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   EtcdSpec   `json:"spec"`
+	Status EtcdStatus `json:"status"`
+}
+
+type EtcdSpec struct {
+	StaticPodOperatorSpec `json:",inline"`
+
+	// forceRedeploymentReason can be used to force the redeployment of the kube-apiserver by providing a unique string.
+	// This provides a mechanism to kick a previously failed deployment and provide a reason why you think it will work
```

Comment:
not sure if godoc will complain about that s/forceRedeploymentReason/ForceRedeploymentReason